src/modules/helpForum.ts

```diff
@@ -157,10 +188,75 @@ export async function helpForumModule(bot: Bot) {
 		},
 	});
 
+	bot.registerCommand({
+		aliases: ['resolved', 'resolve', 'close', 'closed', 'done'],
+		description: 'Help System: Mark a post as resolved',
+		async listener(msg) {
+			changeStatus(msg, true);
+		},
+	});
+
+	bot.registerCommand({
+		aliases: ['reopen', 'open', 'unresolved', 'unresolve'],
+		description: 'Help System: Reopen a resolved post',
+		async listener(msg) {
+			changeStatus(msg, false);
+		},
+	});
+
+	bot.client.on('messageReactionAdd', async reaction => {
+		const message = reaction.message;
+		const thread = await message.channel.fetch();
+		if (!isHelpThread(thread)) {
+			return;
+		}
+		const initial = await thread.fetchStarterMessage();
+		if (initial?.id !== message.id) return;
+		const tag = forumChannel.availableTags.find(
+			t =>
+				t.emoji &&
+				!t.moderated &&
+				t.emoji.id === reaction.emoji.id &&
+				t.emoji.name === reaction.emoji.name,
+		);
+		if (!tag) return;
+		if (thread.appliedTags.length < MAX_TAG_COUNT) {
+			await thread.setAppliedTags([...thread.appliedTags, tag.id]);
+		}
+		await reaction.remove();
+	});
+
+	async function changeStatus(msg: Message, resolved: boolean) {
+		const thread = msg.channel;
+		if (thread?.type !== ChannelType.PublicThread) {
```

Comment:
Won't this check also get threads in #ts-discord-suggestions?

Reply:
True; fixed